x/ibc/applications/transfer/types/keys.go

```diff
@@ -39,11 +42,18 @@ var (
 	DenomTraceKey = []byte{0x02}
 )
 
-// GetEscrowAddress returns the escrow address for the specified channel
-//
-// CONTRACT: this assumes that there's only one bank bridge module that owns the
-// port associated with the channel ID so that the address created is actually
-// unique.
+// GetEscrowAddress returns the escrow address for the specified channel.
+// The escrow address follows the format as outlined in ADR 028:
+// https://github.com/cosmos/cosmos-sdk/blob/master/docs/architecture/adr-028-public-key-addresses.md
 func GetEscrowAddress(portID, channelID string) sdk.AccAddress {
-	return sdk.AccAddress(crypto.AddressHash([]byte(fmt.Sprintf("%s/%s", portID, channelID))))
+	// a slash is used to create domain separation between port and channel identifiers to
+	// prevent address collisions between escrow addresses created for different channels
+	contents := fmt.Sprintf("%s/%s", portID, channelID)
+
+	// ADR 028 AddressHash construction
+	preImage := []byte(EscrowPrefix)
```

Comment:
Hmm what is the precise difference between inserting a prefix here and having a long fixed port name (security-wise)? Is there any?

Reply:
Just a UI thing, if we make the port a long fixed name, then denominations would be long, though it seems apparent this information generally won't be user facing. There is also potential for design misuse. Port identifiers are allowed to be as small as 2 characters.

Should I make the escrow prefix longer?

Reply:
does the length of the prefix increase the difficulty? If not, then we can just use the port as the prefix and the `0` as the separator (ie still using ADR 028 format)

Reply:
I think the length of the prefix does increase the difficulty (of-brute forcing an ECDSA key) the same way a longer port name does. Does ADR 28 analyse this concern in detail? As far as I can tell, it does not, but it should - consider the example of having a 1-character prefix, then every 1/255 (or so) ECDSA keys will have that prefix (although it needs to also have a `/` somewhere, but that's just then 1/255^2 which is still probably feasible).

Reply:
I couldn't find any discussion of this either.

Any recommended prefix length to use?

I think using a prefix separate from the port identifier is a fine solution. It doesn't have any negatives from what I can see and reduces the likelihood of a chain accidentally opening themselves up to an attack

Reply:
I believe even with the current construction on master, using a fixed, long portID makes this attack difficult already. From my understanding, the reason we wanted this feature is to remove any dependence between choice of port-id and security of the escrow address, since this would be non-obvious to a ibc application developer.

Which I agree with. Let's try to reduce the number of quirks developers have to keep in mind. Using a standard length prefix separate from portID makes sense to me.

Reply:
Agreed on the conclusion. I think this may be a point to clarify in ADR 028.